src/cascadia/TerminalApp/TabHeaderControl.cpp

```diff
@@ -38,16 +45,36 @@ namespace winrt::TerminalApp::implementation
         });
     }
 
+    // Method Description
+    // - Retrives the current title
+    // Return Value:
+    // - The current title as a winrt hstring
+    winrt::hstring TabHeaderControl::CurrentHeaderText()
+    {
+        return HeaderTextBlock().Text();
+    }
+
+    // Method Description
+    // - Updates the current title
+    // Arguments:
+    // - The desired title
     void TabHeaderControl::UpdateHeaderText(winrt::hstring title)
     {
         HeaderTextBlock().Text(title);
     }
 
+    // Method Description:
+    // - Updates the zoom icon indicating whether a pane is zoomed
+    // Arguments:
+    // - The desired visibility state of the zoom icon
     void TabHeaderControl::SetZoomIcon(Windows::UI::Xaml::Visibility state)
     {
         HeaderZoomIcon().Visibility(state);
     }
 
+    // Method Description:
+    // - Show the tab rename box for the user to rename the tab title
+    // - We automatically use the previous title as the initial text of the box
```

Comment:
Prefer binding these the other way around, if possible.

I'd prefer you have a property called `Title` that the _Tab may set_, which is data-bound into Xaml.

It's the difference between using xaml _to store the data for you_ (which you are right now) and _storing the data yourself and telling xaml about it_. :smile: